### ISSUE TITLE
Docs: `no-unreachable` is not necessarily handled by TypeScript

### ISSUE THREAD

**Issue by benj-dobs** [eslint/eslint#18378]
### Docs page(s)

https://eslint.org/docs/latest/rules/no-unreachable#handled_by_typescript

### What documentation issue do you want to solve?

The docs currently say:

>  It is safe to disable this rule when using TypeScript because TypeScript's compiler enforces this check. 

This is only true if you have `"allowUnreachableCode": false` in your `tsconfig.json`. The default behaviour is just to surface an editor warning.

https://www.typescriptlang.org/tsconfig#allowUnreachableCode

### What do you think is the correct solution?

If you are using TypeScript, the [`allowUnreachableCode`](https://www.typescriptlang.org/tsconfig#allowUnreachableCode) compiler option offers an alternative to this rule. 

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by nzakas** [comment]
Always happy to make the documentation clearer. Please feel free to open a PR.

**Comment by nzakas** [comment]
@benj-dobs are you planning on submitting a PR for this?

**Comment by benj-dobs** [comment]
Yep - I've been a bit busy but I should get to it this week (assuming nobody else takes it)
